playground/infrastructure/verify.py

```diff
@@ -13,56 +13,90 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-"""
-Module contains the helper for CI step.
-
-It is used to find and verify correctness if beam examples/katas/tests.
-"""
-
+import asyncio
 import logging
 from typing import List
 
-from api.v1.api_pb2 import STATUS_COMPILE_ERROR, STATUS_ERROR, STATUS_RUN_ERROR, \
-    STATUS_RUN_TIMEOUT, \
-    STATUS_VALIDATION_ERROR, STATUS_PREPARATION_ERROR
-from config import Config, Origin
+from api.v1.api_pb2 import Sdk, SDK_PYTHON, SDK_JAVA
+from api.v1.api_pb2 import (
+    STATUS_COMPILE_ERROR,
+    STATUS_ERROR,
+    STATUS_RUN_ERROR,
+    STATUS_RUN_TIMEOUT,
+    STATUS_VALIDATION_ERROR,
+    STATUS_PREPARATION_ERROR,
+)
+from config import Origin, Config
 from grpc_client import GRPCClient
-from helper import Example, get_statuses
-from repository import set_dataset_path_for_examples
+from models import Example, SdkEnum
+from helper import get_statuses
 
 
 class VerifyException(Exception):
-    def __init__(self, error: str):
-        super().__init__()
-        self.msg = error
+    pass
 
-    def __str__(self):
-        return self.msg
 
+class Verifier:
+    """Run examples and verify the results, enrich examples with produced artifacts"""
 
-class CIHelper:
-    """
-    Helper for CI step.
+    _sdk: SdkEnum
+    _origin: Origin
 
-    It is used to find and verify correctness if beam examples/katas/tests.
-    """
+    def __init__(self, sdk: SdkEnum, origin: Origin):
+        self._sdk = sdk
+        self._origin = origin
 
-    async def verify_examples(self, examples: List[Example], origin: Origin):
+    def run_verify(self, examples: List[Example]):
         """
-        Verify correctness of beam examples.
+        Save beam examples and their output in the Google Cloud Datastore.
 
-        1. Find all beam examples starting from directory os.getenv("BEAM_ROOT_DIR")
-        2. Group code of examples by their SDK.
-        3. Run processing for single-file examples to verify examples' code.
+        Outputs for multifile examples are left empty.
         """
-        single_file_examples = list(filter(
-            lambda example: example.tag.multifile is False, examples))
-        set_dataset_path_for_examples(single_file_examples)
-        async with GRPCClient() as client:
-            await get_statuses(client, single_file_examples)
-            await self._verify_examples(client, single_file_examples, origin)
+        single_file_examples = list(
+            filter(lambda example: example.tag.multifile is False, examples)
```

Comment:
@eantyshev Looks like we are doing this two times. First time in _run_ci_cd. Is it intended?

Reply:
Nope, it is redundant, removed
Thanks for pointing out!